### ISSUE TITLE
check custom collection name handling

### ISSUE THREAD

**Issue by FelschR** [FelschR/AspNetCore.Identity.DocumentDB#25]
While having a look at the code today I noticed some strange things being done with the provided CollectionId in the `RegisterDocumentDBStores` methods.

When the CollectionId is provided in the options a DatabaseUri is created from it:
```csharp
// hint: the parameter of CreateDatabaseUri is called "databaseId"
UriFactory.CreateDatabaseUri(dbOptions.CollectionId).ToString()
```
And the underlying RegisterDocumentDBStores method is called with this string as the databaseLink parameter.

In this method the substring is applied on the databaseLink string:
```csharp
var collectionId = databaseLink.Substring(4);
```
And now we are supposed to have a collection ID again? Strange...

Additionally, in the RegisterDocumentDBStores method with documentClient and databaseLink as a parameter, there are a few lines of code that gets the collection and creates it if it did not exist already. And in there, "users" is specifically used as the ID to check for.
This really should be taken from the given collection ID and only use "users" as fallback/default.

I haven't tested anything yet but this seems wrong even if it is only an issue with the variable naming.

Following files should be checked:
* `src/AspNetCore.Identity.DocumentDB/DocumentDBIdentityBuilderExtensions.cs`
* `test/CoreTests/DocumentDBIdentityBuilderExtensionsTests.cs`

**Comment by FelschR** [comment]
Ok, I can confirm that when using options with `AddIdentityWithDocumentDBStores` or `RegisterDocumentDBStores` and setting the CollectionId that a new Database will be created with that name. And the collection always seems to be named "users".

So the current workaround, at least for the database naming issue, is to provide a custom DocumentClient to the DocumentDB provider.

**Comment by FelschR** [comment]
This issue has been fixed in the 2.0 beta.
